pkg/execution/executor/executor.go

```diff
@@ -1487,6 +1508,28 @@ func (e *executor) Execute(ctx context.Context, id state.Identifier, item queue.
 			return nil, err
 		}
 
+		if e.allowStepMetadata.Enabled(ctx, instance.Metadata().ID.Tenant.AccountID) {
+			for _, opcode := range resp.Generator {
+				for _, metadata := range opcode.Metadata {
+					if err := metadata.Validate(); err != nil {
+						l.Warn("invalid metadata in driver response", "error", err)
+						continue
+					}
+
+					// TODO: validate metadata kinds & sizes
```

Comment:
Is this not done with `metadata.Validate()` above?

Reply:
I added a more explicit TODO [271712c](https://github.com/inngest/inngest/pull/3330/commits/271712c0b92eba46a5ce34699d4e81cb614f29e7)